### PR TITLE
SHM #1

### PR DESCRIPTION
**Pull request by kawapa**

### CODE REVIEW

File: shm/inc/Map.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+#include <memory>
+#include <vector>
+
+#include "Coordinates.hpp"
+#include "Island.hpp"
+
+constexpr int minCoordinate = 0;
+constexpr int maxCoordinate = 100;
+constexpr int numberOfIslands = 10;
```
Comment (nauka-programowania-MA) [inline shm/inc/Map.hpp]: not needed here. Move to .cpp

File: shm/inc/Map.hpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class Map {
+public:
+    Island* getIsland(const Coordinates& coordinate);
+
+    Map();
```
Comment (nauka-programowania-MA) [inline shm/inc/Map.hpp:11]: Constructor should be on the top

File: shm/inc/Ship.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    size_t getSpeed() const;
+    size_t getMaxCrew() const;
+    size_t getCapacity() const;
+    void setName(const std::string& name);
+    Cargo* getCargo(size_t index) const;
+    size_t getCargoWeight() const;
+
+    Ship();
+    Ship(int id, const std::string& name, size_t speed, size_t maxCrew, size_t capacity);
+    Ship(int id, size_t speed, size_t maxCrew);
```
Comment (nauka-programowania-MA) [inline shm/inc/Ship.hpp]: move on top

File: shm/src/Coordinates.cpp [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#include "../inc/Coordinates.hpp"
```
Comment (nauka-programowania-MA) [inline shm/src/Coordinates.cpp]: "Coordinates.hpp" Should be enough. 

Terrible practice is add sth like `../`

File: shm/src/Map.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
+#include "../inc/Map.hpp"
+
+#include <algorithm>
+#include <memory>
+#include <random>
+#include <vector>
+
+#include "../inc/Coordinates.hpp"
+#include "../inc/Island.hpp"
```
Comment (nauka-programowania-MA) [inline shm/src/Map.cpp]: same here

File: shm/src/Map.cpp [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    });
+    if (search != end(islands_)) {
+        return search->get();
+    }
+    return nullptr;
+}
+
+std::random_device rd;
+std::mt19937 gen(rd());
+std::uniform_int_distribution<> distrib(minCoordinate, maxCoordinate);
```
Comment (nauka-programowania-MA) [inline shm/src/Map.cpp]: this shouldn't be global variable. Move to function

File: shm/src/Player.cpp [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#include "../inc/Player.hpp"
```
Comment (nauka-programowania-MA) [inline shm/src/Player.cpp]: same here

File: shm/src/Ship.cpp [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+#include "../inc/Cargo.hpp"
+
+size_t Ship::getId() const { return id_; }
+std::string Ship::getName() const { return name_; }
+size_t Ship::getSpeed() const { return speed_; }
+size_t Ship::getMaxCrew() const { return maxCrew_; }
+size_t Ship::getCapacity() const { return capacity_; }
+void Ship::setName(const std::string& name) { name_ = name; }
+Cargo* Ship::getCargo(size_t index) const { return cargo_[index].get(); }
```
Comment (nauka-programowania-MA) [inline shm/src/Ship.cpp]: Validate index

File: shm/src/Ship.cpp [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+Ship::Ship() { id_ = -1; }
+Ship::Ship(int id, const std::string& name, size_t speed, size_t maxCrew, size_t capacity)
+    : id_(id), name_(name), speed_(speed), maxCrew_(maxCrew), capacity_(capacity) { }
+Ship::Ship(int id, size_t speed, size_t maxCrew)
+    : Ship(id, "", speed, maxCrew, 0) { }
+
+Ship& Ship::operator+=(const int crew) {
+    if (crew_ + crew <= maxCrew_) {
+        crew_ += crew;
+    }
```
Comment (nauka-programowania-MA) [inline shm/src/Ship.cpp:28]: There should be some information about unsuccessful operation.

File: shm/src/Ship.cpp [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    return *this;
+}
+
+Ship& Ship::operator-=(const int crew) {
+    if (crew_ > crew) {
+        crew_ -= crew;
+    }
```
Comment (nauka-programowania-MA) [inline shm/src/Ship.cpp:38]: There should be some information about unsuccessful operation.

**Review by nauka-programowania-MA — APPROVED** [review]

**Comment by nauka-programowania-MA** [comment]
{
"requirement_ids": [2568, 2568, 2568, 2568, 2568, 2568, 2568, 2568],
"points": [5, 5, 5, 5, 5, 5, 5, 5]
}

File: shm/src/Player.cpp [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    : ship_(ship), money_(money), availableSpace_(availableSpace) { }
+
+Ship* Player::getShip() const { return ship_.get(); }
+size_t Player::getMoney() const { return money_; }
+size_t Player::getAvailableSpace() const { return availableSpace_; }
+size_t Player::getSpeed() const { return ship_->getSpeed(); }
+Cargo* Player::getCargo(size_t index) const { return ship_->getCargo(index); }
+
+size_t Player::calculateAvailableSpace() {
+    return (ship_->getCapacity() - ship_->getCargoWeight());
```
Comment (WBigaj) [inline shm/src/Player.cpp:15]: Isn't better to do this calculation inside ship and access ship once from player class?

**Review by WBigaj — COMMENTED** [review]
I have checked your code and I have one question.

File: shm/src/Player.cpp (hunk 11 above)
Comment (kawapa) [reply shm/src/Player.cpp:15]: Yes, that makes sense :)

**Pull request closed without merging**